utils.go

```diff
@@ -88,11 +89,16 @@ func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 		return runtimeDir, nil
 	}
 	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+	initCommand, err := ioutil.ReadFile("/proc/1/comm")
+	if err != nil || string(initCommand) != "systemd" {
+		// Non systemd systems follows standard https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html.
+		// /run/user/X can't survive logout, so it is possible to use temp directory only.
+		tmpDir = fmt.Sprintf("%s/containers-user-%d", os.TempDir(), rootlessUID)
+	}
```

Comment:
I had in mind something like:

```
@@ -87,12 +88,15 @@ func getRootlessRuntimeDir(rootlessUID int) (string, error) {
        if err == nil {
                return runtimeDir, nil
        }
-       tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
-       st, err := system.Stat(tmpDir)
-       if err == nil && int(st.UID()) == os.Getuid() && st.Mode()&0700 == 0700 && st.Mode()&0066 == 0000 {
-               return tmpDir, nil
+       initCommand, err := ioutil.ReadFile("/proc/1/comm")
+       if err != nil || string(initCommand) != "systemd" {
+               tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+               st, err := system.Stat(tmpDir)
+               if err == nil && int(st.UID()) == os.Getuid() && st.Mode()&0700 == 0700 && st.Mode()&0066 == 0000 {
+                       return tmpDir, nil
+               }
        }
-       tmpDir = fmt.Sprintf("%s/%d", os.TempDir(), rootlessUID)
+       tmpDir := fmt.Sprintf("%s/containers-user-%d", os.TempDir(), rootlessUID)
```

Does it work for you?  If so, please amend my change in your patch

Reply:
@giuseppe Did you mean `== "systemd"`?

Reply:
Ye this logic is backwards 
if err == nil && string(initCommand) == "systemd" {

Reply:
I think Giuseppe wanted `if err != nil || … == "systemd"`, to fall back to the Systemd logic.

Reply:
@mtrmac yes correct, I want that so we use systemd either on errors or if it is explicitly set.

I'll fix the patch

Reply:
updated, please review.